src/morphing.rs

```diff
@@ -1,6 +1,27 @@
-struct Object {
-    size: usize,
-    path: &'static str,
+use pad::*;
+use objects::*;
+use parsing::{parse_target_size, parse_objects};
+
+
+/// Do ALPaCA's morphing.
+///
+/// If the input object is an HTML page, it samples a new page, changes the
+/// references to its objects accordingly, and pads it; if it is a different
+/// type of object, it returns the object padded to the specified size.
+#[no_mangle]
+pub extern fn morph_object(object: &[u8], request: &str) -> *const u8 {
+    let mut object = Object::from(object, request);
+
+    if object.kind == ObjectKind::HTML {
+        morph_html(&mut object);
+    }
+    else {
```

Comment:
Since we're going to pad the HTML as well, shouldn't these lines (18-22) not be in an else block?

Reply:
Thanks.
`parse_target_size()` takes as input the client's request, and returns the size specified by the GET parameter `alpaca-padding`.
In the case of object type HTML this parameter will not be specified, and just removing the else block wouldn't work.
https://github.com/camelids/libalpaca/blob/c44637618d53b73ef904f4e6c2a8fc204d4229e1/design/specs.txt#L38

My idea was to call `object.pad()` from within `morph_html()`.
I'll push an alternative: `morph_html()` returns the target size for the HTML object.
Still, `parse_target_size()` will need to be within the else block.

Reply:
I like this version a little better. I was thinking we would run `parse_target_size()` on HTML as well, but since we wouldn't want to include an Alpaca query string in a HTML URL (so that the same HTML isn't morphed to the same size repeatedly via some hyperlink posted elsewhere), if we were to run `parse_target_size()` on HTML we'd probably want an `if`/`else` block first thing in `parse_target_size()`, which doesn't make as much sense when we've already separated it out in the calling function like you've done.